examples/scm-provider-generator/scm-provider-example.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: guestbook
+spec:
+  generators:
+  - scmProvider:
```

Comment:
When I apply the example under `examples/scm-provider-generator/scm-provider-example.yaml`, it works :tada: , but (at least on my version of) Argo CD it returns an error within the Application status:

`kubectl apply -f scm-provider-example.yaml`

```yaml
# [jgw@localhost docs]$ k get application/argocd-example-apps-guestbook -o yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  creationTimestamp: "2021-05-09T23:56:20Z"
  name: argocd-example-apps-guestbook
  namespace: argocd-e2e
  # (...)
spec:
  destination:
    namespace: guestbook
    server: https://kubernetes.default.svc
  project: default
  source:
    path: guestbook
    repoURL: git@github.com:argoproj/argocd-example-apps.git
    targetRevision: master
status:
  conditions:
  - lastTransitionTime: "2021-05-09T23:56:20Z"
    message: 'rpc error: code = Unknown desc = error creating SSH agent: "SSH agent requested but SSH_AUTH_SOCK not-specified"'
    type: ComparisonError
  health:
    status: Healthy
  reconciledAt: "2021-05-09T23:56:20Z"
  # (...)
```

This message, `SSH agent requested but SSH_AUTH_SOCK not-specified`, appears to be Argo CD's (somewhat misleading) way of saying that it doesn't have `git@github.com:argoproj/argocd-example-apps.git` added as a private repository (with SSH private key). Once I add the repository it works as expected.

However, since this `scm-provider-example.yaml` is a simple example, we probably want it to work "out of the box" for users (eg without them needing to go through SSH repo add steps), so I recommend adding a cloneProtocol of https to the example, which should alleviate this issue:
```
  - scmProvider:
      cloneProtocol: https
```
Any downside of doing so that you can thing of? Have you seen similar behaviour?

Reply:
D'oh, good catch. Downside of having a global key set up in my dev environment :)